adapters/dab-adb-bridge/src/lib/dab/dab_device_interface.ts

```
@@ -16,13 +16,37 @@
 import { MqttClient } from '../mqtt_client/index.js';
 import  * as topics  from './dab_topics.js';
 import { readFileSync } from 'fs';
+import {
+    AdbBridgeLaunchApplicationRequest,
+    ExitApplicationRequest, GetApplicationStateRequest,
+    KeyPressRequest, LongKeyPressRequest,
+    SetLanguageRequest,
+    StartApplicationTelemetryRequest, StartDeviceTelemetryRequest,
+    StopApplicationTelemetryRequest
+} from "./dab_requests";
+import {
+    DabResponse,
+    DeviceInformationResponse, GetApplicationStateResponse, KeyPressResponse,
+    ListApplicationsResponse, RestartResponse, StartApplicationTelemetryResponse,
+    StartDeviceTelemetryResponse, StopApplicationTelemetryResponse, StopDeviceTelemetryResponse,
+    VersionResponse, HealthCheckResponse
+} from "./dab_responses";
+
+export type NotificationLevel = "info" | "warn" | "debug" | "trace" | "error";
+
+interface Telemetry {
+    device?: NodeJS.Timer;
+    [appId: string]: NodeJS.Timer | undefined;
+}
 
-export class DabDeviceInterface {
+export abstract class DabDeviceInterface {
+    private readonly telemetry: Telemetry;
+    private client!: MqttClient;
```

Is the "!" needed because `client` is only created in `init` and can be undefined? I'd rather throw an explicit error in case `init` has not been called, rather than assume that it is always called. But if it requires too many changes never mind